### PR TITLE
9125 mutually exclusive exception text

### PR DESCRIPTION
**Pull request by wiml**
## Contributor Checklist:

* [X] There is [an associated ticket in Trac](https://twistedmatrix.com/trac/ticket/9125)
* [X] The changes pass minimal style checks
* [X] I have created a newsfragment in src/twisted/newsfragments/
* [X] I have updated the automated tests.

### CODE REVIEW

File: src/twisted/newsfragments/9125.misc [hunk 1]
```diff
@@ -0,0 +1 @@
+Improved the exception text for mutually exclusive function arguments.
```
Comment (exarkun) [inline src/twisted/newsfragments/9125.misc:1]: misc fragments should be empty.  their text won't be included in a news file, it'll just be dropped.

if the text were to be kept and included, the fragment should follow the style guidelines from the twisted wiki.  roughly, "twisted.internet.ssl.CertificateOptions now raises an exception with better text when passed mutually exclusive arguments."

Comment (exarkun) [reply src/twisted/newsfragments/9125.misc:1]: And in this case, the change is actually an improvement to `twisted.python.deprecate`, so making the fragment text about the API there probably makes sense too.  (CertificateOptions could be given as an example beneficiary.)

**Review by exarkun — APPROVED** [review]
Thanks.  The implementation change looks good.  Just one minor comment about the news fragment (inline).

File: src/twisted/newsfragments/9125.misc (hunk 1 above)
Comment (wiml) [reply src/twisted/newsfragments/9125.misc:1]: Ah, I see. Do you think it's worth replacing this newsfragment with an empty one, even though it'll make no difference in the long run (once the newsfragment is deleted after the next release)? Or change it to a non-`.misc` newsfragment? Or should I merge it as-is?

Comment (exarkun) [reply src/twisted/newsfragments/9125.misc:1]: I don't think it will hurt to merge it as-is and then you don't have to wait for another round of CI.

Comment (wiml) [reply src/twisted/newsfragments/9125.misc:1]: Good point. Merged. :)

**Pull request merged**